pinot-common/src/main/java/com/linkedin/pinot/common/partition/ReplicaGroupPartitionAssignment.java

```diff
@@ -0,0 +1,134 @@
+import java.util.NoSuchElementException;
+import java.util.Set;
+
+/**
+ * Class to represent a replica group partition assignment
+ *
+ */
+public class ReplicaGroupPartitionAssignment extends PartitionAssignment {
+  private static final String SEPARATOR = "_";
+
+  public ReplicaGroupPartitionAssignment(String tableName) {
+    super(tableName);
+  }
+
+  public ReplicaGroupPartitionAssignment(String tableName, Map<String, List<String>> partitionToInstances) {
+    super(tableName, partitionToInstances);
+  }
+
+  /**
+   * Compute total number of partitions in the mapping.
+   *
+   * @return the number of partitions
+   */
+  public int getNumPartitions() {
+    Set<String> partitions = new HashSet<>();
+    for (String key : getPartitionToInstances().keySet()) {
+      String partition = key.split(SEPARATOR)[0];
+      partitions.add(partition);
+    }
+    return partitions.size();
+  }
+
+  /**
+   * Compute total number of replica groups in the mapping.
+   *
+   * @return the number of replica groups
+   */
+  public int getNumReplicaGroups() {
+    Set<String> partitions = new HashSet<>();
+    for (String key : getPartitionToInstances().keySet()) {
+      String partition = key.split(SEPARATOR)[1];
+      partitions.add(partition);
+    }
+    return partitions.size();
+  }
+
+  /**
+   * Add an instance to a replica group for a partition.
+   *
+   * @param partition Partition number
+   * @param replicaGroup Replica group number
+   * @param instanceName Name of an instance
+   */
+  public void addInstanceToReplicaGroup(int partition, int replicaGroup, String instanceName) {
+    String key = createMappingKey(partition, replicaGroup);
+    if (!getPartitionToInstances().containsKey(key)) {
+      addPartition(key, new ArrayList<String>());
+    }
+    getInstancesListForPartition(key).add(instanceName);
+  }
+
+  /**
+   * Get instances of a replica group for a partition.
+   *
+   * @param partition Partition number
+   * @param replicaGroup Replica group number
+   * @return List of instances belongs to the given partition and replica group
+   */
+  public List<String> getInstancesfromReplicaGroup(int partition, int replicaGroup) {
+    String key = createMappingKey(partition, replicaGroup);
+    if (!getPartitionToInstances().containsKey(key)) {
+      throw new NoSuchElementException();
+    }
+    return getInstancesListForPartition(key);
+  }
+
+  /**
+   * Set instances of a replica group for a partition.
+   *
+   * @param partition Partition number
+   * @param replicaGroup Replica group number
+   * @param instances Instances that belongs to the gibn partition and replica group
+   */
+  public void setInstancesToReplicaGroup(int partition, int replicaGroup, List<String> instances) {
+    String key = createMappingKey(partition, replicaGroup);
+    addPartition(key, instances);
+  }
+
+  /**
+   * Get all instances.
+   *
+   * @return Set of all instances for this table
+   */
+  public List<String> getAllInstances() {
```

Comment:
This might be useful to have in the super class instead of here

Reply:
moved